### ISSUE TITLE
doubt in output_shape of lambda layer

### ISSUE THREAD

**Issue by shwetagarg-dev** [shyamupa/snli-entailment#3]
I am a bit confused about dimensions. As per my understanding from [here](https://github.com/shyamupa/snli-entailment/blob/master/amodel.py#L99-L102), you will get output of dimension (nb_samples, timesteps, 2 \* opts.lstm_units). Then in this [line](https://github.com/shyamupa/snli-entailment/blob/master/amodel.py#L103) you are choosing last element from time dimension, which should result in an output dimension of (nb_samples, 2 \* opts.lstm_units), but you mentioned output_shape=(k,). 

It would be really nice if you can help me understand that. Thanks

**Comment by shyamupa** [comment]
Copying reply from email,
I think in keras whenever you specify output shape for a layer, you omit the "batch" dimension. So, a softmax layer with output over 300 classes, is actually of shape (NB,300). Hope that makes things easier to understand.

Personally, I feel this makes things a little hard to follow, but that is how keras is designed (as far as I know).
